### PR TITLE
Ls 456 auth

### PR DESCRIPTION
**Pull request by mendonk**
first draft

### CODE REVIEW

File: modules/ROOT/pages/auth.adoc [hunk 1]
```diff
@@ -0,0 +1,170 @@
+= Luna Streaming Authentication
+
+== Authentication
+
+The Helm chart can enable token-based authentication for your Pulsar cluster. For information on token-based
+authentication in Pulsar, go https://pulsar.apache.org/docs/en/security-token-admin/[here].
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:6]:
```suggestion
The Helm chart can enable token-based authentication for your Pulsar cluster. For more, see https://pulsar.apache.org/docs/en/security-token-admin/[Pulsar token authentication].
```

File: modules/ROOT/pages/auth.adoc [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+For authentication to work, the token-generation keys need to be stored in Kubernetes secrets along with some default tokens (for superuser access).
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:8]: What does superuser apply to? Is it required for any authentication? Or are the default tokens used for superuser access?

File: modules/ROOT/pages/auth.adoc [hunk 3]
```diff
@@ -0,0 +1,170 @@
+= Luna Streaming Authentication
+
+== Authentication
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:3]:
```suggestion
```

Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:3]: Move headings up a level b/c everything is in authentication, which is this entire topic.

File: modules/ROOT/pages/auth.adoc [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+
+The Helm chart can enable token-based authentication for your Pulsar cluster. For information on token-based
+authentication in Pulsar, go https://pulsar.apache.org/docs/en/security-token-admin/[here].
+
+For authentication to work, the token-generation keys need to be stored in Kubernetes secrets along with some default tokens (for superuser access). 
+
+The Helm chart includes tooling to automatically create the necessary secrets, or you can do this manually.
+
+=== Automatic generation of secrets for Pulsar token authentication
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:12]:
```suggestion
== Automatic generation of secrets for Pulsar token authentication
```

File: modules/ROOT/pages/auth.adoc [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+
+Use the following settings to enable automatic generation of the secrets and enable token-based authentication:
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:14]: Where is this updated? What is the yaml name?

File: modules/ROOT/pages/auth.adoc [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+----
+enableTokenAuth: yes
+autoRecovery:
+  enableProvisionContainer: yes
+----
+
+When the provision container is enabled, it will check if the required secrets exist. If they don't exist, it will generate new token keys and use those keys to generate the default set of tokens.
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:22]: How do I get from enabling > secrets > tokens? What do I do with the tokens next?

File: modules/ROOT/pages/auth.adoc [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+Using these keys, it will generate tokens for each role listed in `superUserRoles`. Based on the default settings, the following secrets will be created to store the tokens:
+
+* `token-superuser`
+* `token-admin`
+* `token-proxy`
+* `token-websocket`
+
+=== Manual secret creation for Pulsar token authentication
+
+A number of values need to be stored in secrets prior to enabling token-based authentication. First, you need to generate a key-pair for signing the tokens using the Pulsar tokens command:
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:38]:
```suggestion
A number of values need to be stored in secrets prior to enabling token-based authentication.

. Generate a key-pair for signing the tokens using the Pulsar tokens command:
```

File: modules/ROOT/pages/auth.adoc [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+
+`bin/pulsar tokens create-key-pair --output-private-key my-private.key --output-public-key my-public.key`
+
+TIP: The names of the files used in this section match the default values in the chart. If you used different names, then you will have to update the corresponding values.
+
+Then you need to store those keys as secrets.
+
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:45]:
```suggestion
+
----
bin/pulsar tokens create-key-pair --output-private-key my-private.key --output-public-key my-public.key
----
+
[TIP]
====
The names of the files used in this section match the default values in the chart. If you used different names, then you will have to update the corresponding values.
====

. Store those keys as secrets.
+
```

File: modules/ROOT/pages/auth.adoc [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+----
+kubectl create secret generic token-<subject> \
+ --from-file=<subject>.jwt \
+ --namespace pulsar
+----
+
+Once you have created the required secrets, you can enable token-based authentication with this setting in the values:
+
+`enableTokenAuth: yes`
+
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:80]:
```suggestion
+
----
kubectl create secret generic token-public-key \
 --from-file=my-public.key \
 --namespace pulsar
----

. Using those keys, generate tokens for the following subjects (roles): 
+
* `admin`
* `superuser`
* `proxy`
* `websocket` (only required if using the standalone WebSocket proxy)
+
----
bin/pulsar tokens create --private-key file:///pulsar/token-private-key/my-private.key --subject <subject>
----

. Add each token as a secret:
+
----
kubectl create secret generic token-<subject> \
 --from-file=<subject>.jwt \
 --namespace pulsar
----

. Enable token-based authentication with this setting in `devvalues.yaml:
+
----
enableTokenAuth: yes
----

```

**Review by jgillenwater — CHANGES_REQUESTED** [review]
Please see comments, make changes to remaining docs, and resubmit.

File: modules/ROOT/pages/auth.adoc [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+  # caCertificate: |
+----
+
+Once you have created the secrets that store the certificate info (or manually specified it in https://github.com/datastax/pulsar-helm-chart/blob/master/examples/dev-values-keycloak-auth.yaml[values]), enable TLS in the values:
+
+`enableTls: yes`
+
+== Token Authentication via Keycloak Integration
+
+DataStax created the https://github.com/datastax/pulsar-openid-connect-plugin[Pulsar OpenID Connect Authentication Plugin] to provide a more dynamic authentication option for Pulsar. This plugin integrates with Keycloak to dynamically retrieve Public Keys from a running Keycloak instance for token validation. This dynamic public key retrieval enables support for key rotation and multiple authentication/identity providers by configuring multiple allowed token issuers. It also means that token secret keys will *not* be stored in Kubernetes secrets.
```
Comment (michaeljmarshall) [inline modules/ROOT/pages/auth.adoc:118]: Nit: the plugin is supposed to integrate with any OpenID Connect compliant Identity Provider. We use Keycloak as an example, but the spec should allow us to integrate with Okta, Azure AD, and other providers. We haven't not yet tested with these providers, so there is a chance that small adjustments will need to be made.

File: modules/ROOT/pages/auth.adoc [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+----
+
+== TLS
+
+=== Automatically generating certificates using cert-manager
+
+To automatically generate certificates using cert-manager, see https://cert-manager.io/docs/[Cert-Manager Documentation].
+
+=== Manually configuring certificate secrets for TLS
+
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:94]: Consider making this a partial and add into `quickstart-helm-installs.adoc`.

File: modules/ROOT/pages/auth.adoc [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+--auth-plugin "org.apache.pulsar.client.impl.auth.oauth2.AuthenticationOAuth2" 
+--auth-params '{"privateKey":"file:///pulsar/creds.json","issuerUrl":"http://test-keycloak:8081/auth/realms/pulsar","audience":"I dont matter"}' 
+--tenants list 
+public
+pulsar
+----
+
+You're now using Keycloak tokens with `pulsar-admin` CLI.
+
+== Next
```
Comment (jgillenwater) [inline modules/ROOT/pages/auth.adoc:210]:
```suggestion
== What's next?
```

**Review by jgillenwater — APPROVED** [review]
Minor suggestion

**Pull request merged**